src/analysis/schwartz_space.lean

```diff
@@ -403,6 +405,43 @@ begin
   rwa [pow_zero, one_mul] at this,
 end
 
+lemma le_sup_seminorm {k n k' n' : ℕ} (hk : k' ≤ k) (hn : n' ≤ n) :
+  (seminorm 𝕜 k' n' : seminorm 𝕜 𝓢(E, F)) ≤ (finset.Iic (k, n)).sup (λ n, seminorm 𝕜 n.1 n.2) :=
+@finset.le_sup _ _ _ _ _ (λ n : ℕ × ℕ, seminorm 𝕜 n.1 n.2) _ $ finset.mem_Iic.2 $
+  prod.mk_le_mk.2 ⟨hk, hn⟩
```

Comment:
Are you sure this doesn't Just Work? I suspect you thought you needed this lemma because you hit unification errors that we now solved.
```suggestion
finset.le_sup $ finset.mem_Iic.2 $ prod.mk_le_mk.2 ⟨hk, hn⟩
```

Reply:
nope, this fails with
````
type mismatch, term
  finset.le_sup (finset.mem_Iic.mpr (prod.mk_le_mk.mpr ⟨hk, hn⟩))
has type
  ?m_3 (k', n') ≤ (finset.Iic (k, n)).sup ?m_3
but is expected to have type
  schwartz_map.seminorm 𝕜 k' n' ≤
    (finset.Iic (k, n)).sup (λ (n : ℕ × ℕ), schwartz_map.seminorm 𝕜 n.fst n.snd)
````

Reply:
Still, I'm not sure this lemma is worth having. What do you think of inlining it in the proof of `le_sup_seminorm_apply`?

Reply:
I did not get inlining to work either. There is some serious abuse of defeq going on here.

Reply:
This works:
```suggestion
eq.trans_le (by refl) (finset.le_sup $ finset.mem_Iic.2 $ prod.mk_le_mk.2 ⟨hk, hn⟩)
```

Reply:
The idiomatic solution in other parts of the library is adding the lemma `finset.le_sup_of_le`. See eg [`le_supr_of_le`](https://leanprover-community.github.io/mathlib_docs/order/complete_lattice.html#le_supr_of_le).

Reply:
Perhaps we should drop this lemma entirely, and add
```lean
lemma finset.le_sup_uncurry {α β γ} [semilattice_sup γ] [order_bot γ]
  (f : α → β → γ) {a : α} {b : β} {s : finset (α × β)} (hs : (a, b) ∈ s) :
  f a b ≤ s.sup (function.uncurry f) :=
(function.uncurry_apply_pair _ _ _).symm.trans_le $ finset.le_sup hs
```
instead

Reply:
I think my `le_sup_of_le` subsumes your lemma even.